parsec/parsec.c

```diff
@@ -1583,8 +1598,57 @@ parsec_update_deps_with_counter(const parsec_taskpool_t *tp,
     return dep_cur_value == 0;
 }
 
-static int
-parsec_update_deps_with_mask(const parsec_taskpool_t *tp,
+int
+parsec_update_deps_with_counter_count_task(parsec_taskpool_t *tp,
+                                           const parsec_task_t* restrict task,
+                                           parsec_dependency_t *deps,
+                                           const parsec_task_t* restrict origin,
+                                           const parsec_flow_t* restrict origin_flow,
+                                           const parsec_flow_t* restrict dest_flow)
+{
+    parsec_dependency_t dep_new_value, dep_cur_value;
+#if defined(PARSEC_DEBUG_PARANOID) || defined(PARSEC_DEBUG_NOISIER)
+    char tmp[MAX_TASK_STRLEN];
+    parsec_task_snprintf(tmp, MAX_TASK_STRLEN, task);
+#endif
+
+    (void)origin;
```

Comment:
same as above

Reply:
Some termination detectors (e.g. Credit Distribution Detection) need to inject additional data inside the messages themselves, this part of the API allows this.